### PR TITLE
updated code according to feedback, removed addUser() and added more …

### PR DESCRIPTION
**Pull request by HappyJuice123**
…tests

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -163,7 +163,7 @@ describe("app", () => {
     });
     test("201: POST - adds an object onto comments with the properties username and body and responds with the posted comment", () => {
       const newComment = {
-        username: "HappyJuice123",
+        username: "mallionaire",
```
Comment (JVal-NC) [inline __tests__/app.test.js:291]: Nice work on fixing these up, another test for posting a comment with a nonexistent user would be good.

File: db/Models/app.models.js [hunk 2]
```diff
@@ -60,27 +60,6 @@ function fetchComments(review_id) {
...
-      ]
-    )
-    .then(({ rows }) => {
-      return rows[0];
-    });
-}
-
 function addingComment(review_id, username, body) {
   return db
     .query(
```
Comment (JVal-NC) [inline db/Models/app.models.js:133]: Good query but no need to insert new date/votes, SQL should handle this for us.

File: db/Controllers/app.controllers.js [hunk 3]
```diff
@@ -61,13 +60,18 @@ function addComment(req, res, next) {
...
-  const addUserPromise = addUser(username);
-  const addCommentPromise = addingComment(review_id, username, body);
+  const reviewIdPromise = fetchReview(review_id);
+  const addingCommentPromise = addingComment(review_id, username, body);
 
-  Promise.all([addUserPromise, addCommentPromise]).then((result) => {
-    const addComment = result[1];
-    res.status(201).send({ addComment });
-  });
+  Promise.all([reviewIdPromise, addingCommentPromise])
```
Comment (JVal-NC) [inline db/Controllers/app.controllers.js]: ReviewIdPromise unnecessary, SQL will check foreign keys for us and return an error. Have a look at the seed file to see which these are if your unsure.

**Review by JVal-NC — COMMENTED** [review]
Couple comments to incorporate for a new PR

File: __tests__/app.test.js [hunk 4]
```diff
@@ -227,18 +227,74 @@ describe("app", () => {
...
+        body: "i like cats",
+      };
+
+      return request(app)
+        .post("/api/reviews/50000/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("Bad Request");
```
Comment (JVal-NC) [inline __tests__/app.test.js:360]: Is this a bad request or just not found?

Also, is there a test for a review_id that isnt a number?

File: __tests__/app.test.js [hunk 5]
```diff
@@ -227,18 +227,74 @@ describe("app", () => {
...
+    test("400: POST - responds with msg when sent a query with a username that does not exist", () => {
+      const newComment = {
+        username: "hello123",
+        body: "i like cats",
+      };
+
+      return request(app)
+        .post("/api/reviews/1/comments")
+        .send(newComment)
+        .expect(400)
```
Comment (JVal-NC) [inline __tests__/app.test.js]: I'd suggest this a 404 not found

File: db/Models/app.models.js [hunk 6]
```diff
@@ -60,40 +60,22 @@ function fetchComments(review_id) {
...
       VALUES
-      ($1, $2, $3, $4, (SELECT username FROM users WHERE username = $5))
+      ($1, $2, $3)
       RETURNING *
       `,
-      [body, 0, review_id, new Date(1676994323752), username]
+      [body, review_id, username]
     )
     .then(({ rows }) => {
+      if (!rows) {
```
Comment (JVal-NC) [inline db/Models/app.models.js]: What is this doing? SQL will throw an error if it doesn't insert something for the test cases. Suspect this isn't needed.

**Review by JVal-NC — COMMENTED** [review]
Jason, few comments to consider. Happy to have a call if anything is unclear

File: __tests__/app.test.js [hunk 7]
```diff
@@ -227,20 +227,90 @@ describe("app", () => {
...
       return request(app)
-        .get("/api/reviews/50000/comments")
+        .post("/api/reviews/50000/comments")
+        .send(newComment)
         .expect(404)
         .then(({ body }) => {
           expect(body.msg).toBe("Not Found");
         });
     });
+    test("404: POST - responds with msg when sent a query with a non-valid review_id", () => {
```
Comment (pippaamy) [inline __tests__/app.test.js]: remember to change the description of that status code

**Review by pippaamy — APPROVED** [review]
Nice work, just don't forget to change the status code in descriptions when you're changing tests!

**Pull request merged**